omim2obo/main.py

```diff
@@ -280,49 +280,61 @@ def omim2obo(use_cache: bool = False):
+
+    # - Disease-Gene relations, 2 of 2: Add relations (subclass restrictions)
+    #   Only add if 1 association, or 2 associations and phenotype is digenic
+    for p_mim, assocs in phenotype_genes.items():
+        p_lab: str = assocs[0]['phenotype_label']
+        is_digenic = 'digenic' in p_lab.lower()
+        if is_digenic and len(assocs) == 1:
+            print('Unexpected only 1 gene-phenotype association found in digenic phenotype:', p_mim, p_lab,
+                file=sys.stderr)
+        if is_digenic and len(assocs) > 2:
+            print('Unexpected >2 gene-phenotype associations found in digenic phenotype:', p_mim, p_lab,
+                file=sys.stderr)
+        if len(assocs) > 1 and not is_digenic:
+            continue
         for assoc in assocs:
-            # p_lab currently not used
-            p_mim, p_lab, p_map_key, p_map_lab = \
-                assoc['phenotype_mim_number'], assoc['phenotype_label'], assoc['phenotype_mapping_info_key'], \
-                assoc['phenotype_mapping_info_label']
-            # Provenance: https://github.com/monarch-initiative/omim/issues/78
-            if not p_mim:
-                continue
-            # Provenance: https://github.com/monarch-initiative/omim/issues/79#issuecomment-1319408780
-            if p_map_key == '1':
-                continue
-
-            # RO:0003302 (causes or contributes to condition)
-            # https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http://purl.obolibrary.org/obo/RO_0003302
-            # Provenance for this decision:
-            # - Multiple rows, same mapping key: https://github.com/monarch-initiative/omim/issues/75
-            # - Multiple rows, diff mapping keys: https://github.com/monarch-initiative/omim/issues/81
-            evidence = f'Evidence: ({p_map_key}) {p_map_lab}'
-            predicate = RO['0003302']  # default if `len(assocs) > 1`
-            if len(assocs) == 1:
-                predicate = MORBIDMAP_PHENOTYPE_MAPPING_KEY_PREDICATES[p_map_key]
-            add_subclassof_restriction_with_evidence(graph, predicate, OMIM[p_mim], OMIM[mim_number], evidence)
-            if predicate in MORBIDMAP_PHENOTYPE_MAPPING_KEY_INVERSE_PREDICATES:
-                # The following code basically just adds the converse relation:
-                # If there is g2d in the previous code, we now, in addition, add d2g
-                inverse_predicate = MORBIDMAP_PHENOTYPE_MAPPING_KEY_INVERSE_PREDICATES[predicate]
-                add_subclassof_restriction_with_evidence(
-                    graph, inverse_predicate, OMIM[mim_number], OMIM[p_mim], evidence)
+            gene_mim, p_lab, p_map_key, p_map_lab = assoc['gene_id'], assoc['phenotype_label'], \
+                assoc['mapping_key'], assoc['mapping_label']
+            # TODO: @Trish: Decide: Should have more relationships now because I simplified to what Sabrina asked.
```

Comment:
## Marking casual (`RO:0004003`)
I'm not sure if this is what we want. Please review.
- a. Sabrina said that if 1 gene assoc w/ disease, it's causal, and we can declare it "RO:0004003 (has material basis in germline mutation in)".
- b. But before, we were only doing that if OMIM specified the gene->disease relation as "'3': 'The molecular basis for the disorder is known; a mutation has been found in the gene.',", which we have associated with RO:0004013 (is causal germline mutation in), and that is the inverse of RO:0004003

Reply:
I think the previous check can remain in place. Confirm that only 1 gene unless digenic and also make sure it has mapping key (3).

Generally I think that any disease that has only gene listed listed in the phenotype-gene table for the entry (or in morbidmap, maybe found in the other files too), will all have mapping key (3) based on they key definition "The molecular basis for the disorder is known; a mutation has been found in the gene.". However, there were entries in the Excel file that had two lines in morbidmap and the mapping keys were different between these lines in morbidmap.

Reply:
@joeflack4 I have analyzed the morbidmap file and there are either a combination of issues with data in that file and/or other edge cases we need to consider to determine if a gene is causal.

Reply:
Decided that yes, we'll only do mapping key = 3 again. Resolving.

Reply:
Also, note my comment about digenic diseases is also no longer valid based on Sabrina's last email (Gene association in Mondo) on the rule or did you interpret her comment that the handful of diseases that have digenic in the phenotype label and also have only 1 gene association and have mapping key 3 should be included?